src/main/java/fi/aalto/cs/apluscourses/dal/APlusTokenAuthentication.java

```diff
@@ -0,0 +1,39 @@
+package fi.aalto.cs.apluscourses.dal;
+
+import org.apache.http.HttpRequest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class APlusTokenAuthentication extends TokenAuthentication {
```

Comment:
`TokenAuthentication` can be any `Authentication` that is based on token of `char[]`. `APlusTokenAuthentication` extends that by information how to use token with A+ API.